### PR TITLE
LPS-198083 Upgrade `webpack` inside `frontend-js-svg4everybody-web`

### PR DESCRIPTION
**Pull request by kresimir-coko**
[Jira ticket](https://liferay.atlassian.net/browse/LPS-196980)

`yarn why webpack` inside `frontend-js-svg4everybody-web` returns `webpack@5.70.0` which is wrong. This is due to the `modules/package.json`'s resolution field specifying `webpack@5.70.0`.

So I added an extra commit that handles the `resolutions` change. I'm not 100% sure that's the right call, but I think it is.

### CODE REVIEW

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by izaera** [comment]
Why do we need to specify any webpack version at all? Aren't `npm-scripts` supposed to run `webpack` with our centralized version when they find a config file for it?

@bryceosterhaus Any idea? :thinking:

**Comment by bryceosterhaus** [comment]
Webpack is defined in that module because of the webpack.config.js, and it requires webpack.

However, this shouldn't be updated in portal, but instead should be fixed in npm-scripts. We likely need to update svg4everybody module so that is just esm and doesn't need it's own webpack config

**Comment by izaera** [comment]
> Webpack is defined in that module because of the webpack.config.js, and it requires webpack.

Nope, because npm-scripts simply invokes webpack if the file is there -> 
https://github.com/liferay/liferay-frontend-projects/blob/master/projects/npm-tools/packages/npm-scripts/src/scripts/build.js#L219-L221

In fact we have a `webpack.config.js` file in `frontend-js-web` too.

> However, this shouldn't be updated in portal, but instead should be fixed in npm-scripts. We likely need to update svg4everybody module so that is just esm and doesn't need it's own webpack config

That's a different thing. I don't know if it is needed or not.

But I'd say that invoking `npm-scripts` build in this project and removing the `webpack` dependency should do the trick.

**Comment by bryceosterhaus** [comment]
> removing the webpack dependency should do the trick.

We can't remove it due to our SF rule... That is what I meant by, it is defined because we use it in the module.

So we should probably make an exception for that by adding "webpack" here, https://github.com/liferay/liferay-portal/blob/master/modules/.eslintrc.js#L33-L44

**Comment by kresimir-coko** [comment]
ci:test:relevant

File: modules/yarn.lock [hunk 1]
```diff
@@ -10265,6 +10265,11 @@ lie@~3.3.0:
   dependencies:
     immediate "~3.0.5"
 
+liferay-ckeditor@4.18.0-liferay.4:
+  version "4.18.0-liferay.4"
+  resolved "https://registry.yarnpkg.com/liferay-ckeditor/-/liferay-ckeditor-4.18.0-liferay.4.tgz#08ddc7c1025f6c036a9de97c3403b9928e6b99dc"
+  integrity sha512-mugcikGLtVbZ1kqxtHjf1z8F4hWPeET4EyCx48ePIUc01fnOuAEiiNwnaWeBxiMS8K75CvYNbA37jJhCskHXmw==
+
```
Comment (bryceosterhaus) [inline modules/yarn.lock:10272]: Why did this get added?

Comment (kresimir-coko) [reply modules/yarn.lock:10272]: no idea, something (running `yarn` in `modules` maybe) added it automatically 🤷

Comment (kresimir-coko) [reply modules/yarn.lock:10272]: I will `yarn why` it

Comment (kresimir-coko) [reply modules/yarn.lock:10272]: @bryceosterhaus `yarn why liferay-ckeditor` inside modules:
```
=> Found "liferay-ckeditor@4.18.0-liferay.4"
info Reasons this module exists
   - "_project_#@liferay#frontend-js-dependencies-web" depends on it
   - Hoisted from "_project_#@liferay#frontend-js-dependencies-web#liferay-ckeditor"
```

Comment (bryceosterhaus) [reply modules/yarn.lock:10272]: ah I see. maybe just didn't get added before 

PR looks fine to me, just need CI or QA sign off

**Comment by kresimir-coko** [comment]
ci:test:relevant

File: modules/yarn.lock (hunk 1 above)
Comment (markocikos) [reply modules/yarn.lock:10272]: We recently [updated CKEditor](https://github.com/liferay-frontend/liferay-portal/pull/3701) to 4.21.0-liferay.1. It looks like we forgot to update all dependencies cc @fortunatomaldonado 

@kresimir-coko can you please update it in `frontend-js-dependencies-web`

Comment (kresimir-coko) [reply modules/yarn.lock:10272]: sure!

Comment (kresimir-coko) [reply modules/yarn.lock]: https://github.com/liferay-frontend/liferay-portal/pull/3724/commits/7714187f5e9aa8cc93483fa3e54dbb0c7046550a good enough? @markocikos

Comment (markocikos) [reply modules/yarn.lock]: Perfect! :sparkles:

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by mateusralv** [comment]
ci:test:stable

**Comment by mateusralv** [comment]
Hey @kresimir-coko I found a failure related to your PR, could you take a look?

https://github.com/liferay-frontend/liferay-portal/assets/44584050/c4172d3d-8f66-496c-9345-0dae4b7ce20c


Test affected: 
LocalFile.BlogsAdmin#CanAddEntryInAdmin

[Testray Comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1697028468911&testrayRunIdA=2438944747&testrayRunIdB=2439262467)

**Comment by kresimir-coko** [comment]
Looking, thanks

**Comment by kresimir-coko** [comment]
@markocikos this looks like a ckeditor thing. I ant all-ed with the commit and without, and on clean master, and it's the ckeditor commit that is the cause of the error.

What should we do? The ckeditor version bump isn't in the scope of the task but it's causing this PR to fail.

**Comment by markocikos** [comment]
@kresimir-coko Please revert the ckeditor version update, and commit artifacts [we discussed above](https://github.com/liferay-frontend/liferay-portal/pull/3724#discussion_r1350195602). 

@fortunatomaldonado I reopened [LPS-196513 CKEditor v4.18 Update](https://liferay.atlassian.net/browse/LPS-196513). We need to update version in `frontend-js-dependencies-web`, and fix the issue in https://github.com/liferay-frontend/liferay-portal/pull/3724#issuecomment-1758511822 as a part of the upgrade. https://github.com/liferay-frontend/liferay-portal/pull/3701 was not enough to bump up version, you can see proof on blogs page:

image

**Comment by kresimir-coko** [comment]
> commit artifacts we talked about

@markocikos Just double-checking, by artifacts you mean this?
```
liferay-ckeditor@4.18.0-liferay.4:
  version "4.18.0-liferay.4"
  resolved "https://registry.yarnpkg.com/liferay-ckeditor/-/liferay-ckeditor-4.18.0-liferay.4.tgz#08ddc7c1025f6c036a9de97c3403b9928e6b99dc"
  integrity sha512-mugcikGLtVbZ1kqxtHjf1z8F4hWPeET4EyCx48ePIUc01fnOuAEiiNwnaWeBxiMS8K75CvYNbA37jJhCskHXmw==
```

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
ci:test:stable

**Review by markocikos — APPROVED** [review]
The part related to CKEditor LGTM!

**Comment by kresimir-coko** [comment]
ci:forward

**Comment by deborasoliveira** [comment]
I'll run relevant again because many tests weren't executed on CI.

**Comment by deborasoliveira** [comment]
ci:test:relevant

**Comment by deborasoliveira** [comment]
Some tests are still not being executed but I'll continue the analysis tomorrow with this recent result.

**Comment by kresimir-coko** [comment]
Have you had the time to check the results? @deborasoliveira

**Comment by deborasoliveira** [comment]
The failures seem to be unrelated. 
[Comparison run](https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=2&testrayRunIdB=2441915214&testrayRunIdA=2442197218&statusB=3)

**Comment by kresimir-coko** [comment]
Manually forwarded to https://github.com/brianchandotcom/liferay-portal/pull/142388

**Pull request closed without merging**